npm_and_yarn/spec/dependabot/npm_and_yarn/dependency_files_filterer_spec.rb

```diff
@@ -14,117 +14,95 @@
   end
 
   let(:dependency_files) do
-    [
-      package_json,
-      yarn_lock,
-      npm_lock,
-      nested_package_json,
-      nested_shrinkwrap
-    ]
+    project_dependency_files(project_name)
   end
-
+  let(:project_name) { "npm6_and_yarn/simple" }
   let(:updated_dependencies) { [dependency] }
-
   let(:dependency) do
     Dependabot::Dependency.new(
-      name: "etag",
-      version: "1.0.0",
+      name: "fetch-factory",
+      version: "0.0.2",
       requirements: [{
         file: "package.json",
-        requirement: "^1.0.0",
-        groups: ["dependencies"],
-        source: nil
-      }],
-      package_manager: "npm_and_yarn"
-    )
-  end
-  let(:nested_dependency) do
-    Dependabot::Dependency.new(
-      name: "chalk",
-      version: "3.10.1",
-      requirements: [{
-        file: "packages/package1/package.json",
-        requirement: "^3.10.1",
+        requirement: "^0.0.1",
         groups: ["dependencies"],
         source: nil
       }],
       package_manager: "npm_and_yarn"
     )
   end
 
-  let(:package_json) do
-    Dependabot::DependencyFile.new(
-      name: "package.json",
-      content: fixture("package_files", "package.json")
-    )
-  end
-  let(:yarn_lock) do
-    Dependabot::DependencyFile.new(
-      name: "yarn.lock",
-      content: fixture("yarn_lockfiles", "yarn.lock")
-    )
-  end
-  let(:npm_lock) do
-    Dependabot::DependencyFile.new(
-      name: "package-lock.json",
-      content: fixture("npm_lockfiles", "package-lock.json")
-    )
-  end
-  let(:nested_package_json) do
-    Dependabot::DependencyFile.new(
-      name: "packages/package1/package.json",
-      content: fixture("package_files", "package1.json")
-    )
-  end
-  let(:nested_shrinkwrap) do
-    Dependabot::DependencyFile.new(
-      name: "packages/package1/npm-shrinkwrap.json",
-      content: fixture("shrinkwraps", "npm-shrinkwrap.json")
-    )
+  def project_dependency_file(file_name)
+    dependency_files.find { |f| f.name == file_name }
   end
 
   describe ".files_requiring_update" do
     it do
-      is_expected.to contain_exactly(package_json, yarn_lock, npm_lock)
+      is_expected.to contain_exactly(
+        project_dependency_file("package.json"),
+        project_dependency_file("package-lock.json"),
+        project_dependency_file("yarn.lock")
```

Comment:
This is so much clearer, thanks

Reply:
Ops just moved this to a different PR to make it easier to review: https://github.com/dependabot/dependabot-core/pull/3148